### PR TITLE
NFL updates

### PR DESCRIPTION
**Pull request by justinbarry**

### CODE REVIEW

**Review by bthaile — APPROVED** [review]
Lgtm

File: packages/composites/augur/src/dataProviders/sportsdataio.ts [hunk 1]
```diff
@@ -64,45 +82,131 @@ interface CommonScores {
...
+        data: {
+          sport,
+          endpoint: 'schedule',
+          season: '2021', // TODO: Find a source for this information.
+        },
+      }
+
+      const response = await exec(input, context)
+      const filtered = (response.result as { GlobalGameID: number }[]).filter(
+        (event) => event.GlobalGameID != 0,
```
Comment (robert-e-davidson3) [inline packages/composites/augur/src/dataProviders/sportsdataio.ts:195]: do you intend to use `!=` not `!==`?

File: packages/composites/augur/src/dataProviders/sportsdataio.ts [hunk 2]
```diff
@@ -142,7 +250,21 @@ const getScores = async (
...
+        data: {
+          sport,
+          endpoint: 'scores',
+          season: '2021', // TODO: Find a source for this information.
+        },
+      }
+
+      const response = await exec(input, context)
+      const filtered = (response.result as { GlobalGameID: number }[]).filter(
+        (event) => event.GlobalGameID != 0,
```
Comment (robert-e-davidson3) [inline packages/composites/augur/src/dataProviders/sportsdataio.ts:264]: also

File: packages/composites/augur/src/methods/createMarkets.ts [hunk 3]
```diff
@@ -52,16 +54,26 @@ export const execute: ExecuteWithConfig<Config> = async (input, context, config)
...
+  sport: string,
+  contractAddress: string,
+  input: AdapterRequest,
+  context: AdapterContext,
+  config: Config,
+) => {
+  const contract =
+    sport === 'nfl'
+      ? new ethers.Contract(contractAddress, NFL_ABI, config.wallet)
+      : new ethers.Contract(contractAddress, TEAM_ABI, config.wallet)
```
Comment (robert-e-davidson3) [inline packages/composites/augur/src/methods/createMarkets.ts:68]:
```suggestion
  const contract = new ethers.Contract(contractAddress, sport === 'nfl' ? NFL_ABI : TEAM_ABI, config.wallet)
```

File: packages/composites/augur/src/methods/resolveMarkets.ts [hunk 4]
```diff
@@ -53,8 +49,18 @@ export const execute: ExecuteWithConfig<Config> = async (input, context, config)
...
-  const contract = new ethers.Contract(contractAddress, TEAM_ABI, config.wallet)
+const resolveTeam = async (
+  jobRunID: string,
+  sport: string,
+  contractAddress: string,
+  context: AdapterContext,
+  config: Config,
+) => {
+  // The difference isn't meaningful here using the proper abis anyway.
+  const contract =
```
Comment (robert-e-davidson3) [inline packages/composites/augur/src/methods/resolveMarkets.ts:60]: also

**Review by robert-e-davidson3 — APPROVED** [review]

File: packages/composites/augur/src/dataProviders/sportsdataio.ts (hunk 1 above)
Comment (justinbarry) [reply packages/composites/augur/src/dataProviders/sportsdataio.ts:195]: That only showed up changed due to formatting. 🤷

File: packages/composites/augur/src/dataProviders/sportsdataio.ts (hunk 2 above)
Comment (justinbarry) [reply packages/composites/augur/src/dataProviders/sportsdataio.ts:264]: That only showed up changed due to formatting. 🤷

**Comment by pgebheim** [comment]
@boxhock -- This is an adapter update to fix many issues with the NFL support. Please take a look.

File: packages/composites/augur/src/dataProviders/sportsdataio.ts [hunk 5]
```diff
@@ -64,45 +82,131 @@ interface CommonScores {
...
       }))
     }
     case 'ncaa-fb': {
-      return (filtered as CFBGames[]).map(event => ({
+      const input = {
+        id,
+        data: {
+          sport,
+          endpoint: 'schedule',
+          season: '2021', // TODO: Find a source for this information.
```
Comment (boxhock) [inline packages/composites/augur/src/dataProviders/sportsdataio.ts:189]: Hard-coding a year seems like a bad idea, as it's easily forgotten. Can we come up with a solution for this before merging?

File: packages/composites/augur/src/dataProviders/sportsdataio.ts [hunk 6]
```diff
@@ -142,7 +250,21 @@ const getScores = async (
       }))
     }
     case 'ncaa-fb': {
-      return (filtered as CFBGames[]).map(event => ({
+      const input = {
+        id,
+        data: {
+          sport,
+          endpoint: 'scores',
+          season: '2021', // TODO: Find a source for this information.
```
Comment (boxhock) [inline packages/composites/augur/src/dataProviders/sportsdataio.ts:258]: Same thing here

File: packages/composites/augur/src/methods/createMarkets.ts [hunk 7]
```diff
@@ -84,18 +96,34 @@ const createTeam = async (jobRunID: string, sport: string, contractAddress: stri
...
-      Math.floor(event.startTime / 1000),
-      Math.round(event.homeSpread*10),
-      Math.round(event.totalScore*10),
-      event.createSpread,
-      event.createTotalScore,
-      event.moneylines,
-      { nonce }
-    ]
+    const payload =
+      sport !== 'nfl'
```
Comment (boxhock) [inline packages/composites/augur/src/methods/createMarkets.ts:100]: Above we're checking if === NFL, while here we're checking !== NFL. Could we be consistent with checking the same thing on both, so it's easier to follow the flow?

File: packages/sources/sportsdataio/src/sport/nfl/index.ts [hunk 8]
```diff
@@ -17,12 +17,18 @@ export const execute: ExecuteWithConfig<Config> = async (request, context, confi
...
+    }
     case schedule.NAME: {
       return await schedule.execute(request, context, config)
     }
     case scores.NAME: {
       return await scores.execute(request, context, config)
     }
+    case teams.NAME: {
+      return await teams.execute(request, context, config)
+    }
```
Comment (boxhock) [inline packages/sources/sportsdataio/src/sport/nfl/index.ts:31]: Can we add documentation of these new endpoints in the README?

**Review by boxhock — CHANGES_REQUESTED** [review]

File: packages/sources/sportsdataio/src/sport/nfl/index.ts (hunk 8 above)
Comment (justinbarry) [reply packages/sources/sportsdataio/src/sport/nfl/index.ts:31]: Done.

**Review by boxhock — APPROVED** [review]

**Pull request merged**